### DISCUSSION TITLE
need a systemd service unit file to run as root and start podman as a regular user

### DISCUSSION THREAD

**Discussion by ebryerwork** [containers/podman#17333] [Q&A] [answered]
I want to create a systemd service unit file run by root that will start a rootless container as licuser1, a regular user. I read in podman-generate-systemd:
```
Once the systemd unit file is generated, install it to /etc/systemd/system to be run by the root user or to $HOME/.config/systemd/user for installing it as a non-root user. Enable the copied unit file or files using systemctl enable.
```
That either requires systemctl to be run as the user or to run podman as root. Does anyone know how I can modify that file or create another one such that it is started with systemctl by root at system startup while it runs podman rootless as licuser1? I tried adding 
User=licuser1
Group=licuser1
to the Service section, but that doesn't work. 

During the xCAT postscript that runs as root during the startup of this system, I issue `loginctl enable-linger $(grep licuser1 /etc/passwd | cut -d: -f3)` and put in licuser1's .bashrc:
```
export XDG_RUNTIME_DIR=/run/user/$(grep licuser1 /etc/passwd | cut -d: -f3)
export DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/$(grep licuser1 /etc/passwd | cut -d: -f3)/bus
```
That should take care of that issue.

**Comment by Luap99** [comment, marked as answer]
`User=`/`Group=` do not work and are not supported by us, you need to run the unit in your systemd user session as it is described in the man page.

If you just want the startup at boot use the enable linger mode for the user as mentioned in the man page.

**Reply by ebryerwork** [reply]
My mistake. I hadn't realized one could, as a regular user, enable a service, and systemd would start that service at boot.

**Reply by eriksjolund** [reply]
There is already a feature request

* https://github.com/containers/podman/issues/12778
